### ISSUE TITLE
BUG: Unnecessary permissions in Helm Charts

### ISSUE THREAD

**Issue by Yseona** [openelb/openelb#421]
### Description
The bug is that the Deployment openelb-manager in the Helm charts has too much RBAC permission than it needs. The service account of `openelb-manager` is bound to a clusterrole with the following permissions:

- `create/patch/update/delete` verb of the `deployments` resource (ClusterRole)
- `patch/update` verb of the `daemonsets` resource (ClusterRole)
- `create/patch/delete` verb of the `services` resource (ClusterRol

After reading the source code of [openelb](https://github.com/openelb/openelb), I didn't find any Kubernetes API usages using these permissions. Besides, some of these unused permissions may have potential risks. For example, if malicious users gain control of a Kubernetes node running a `openelb-manager` pod, they can use the `create deployments` permission to create privileged containers with malicious container images.

Therefore, these permissions should be rechecked to determine if they are truly unnecessary. If they are, the issue should be fixed by removing the unnecessary permissions or other feasible methods. 

**To Reproduce**

Use the helm charts with default values.

`helm repo add stable https://charts.kubesphere.io/stable`
`helm template openelb stable/openelb`

**Comment by renyunkang** [comment]
We do need these rbac permissions, see [source code](https://github.com/openelb/openelb/blob/release-0.5/pkg/controllers/lb/controller.go#L298) .

**Comment by kaaass** [comment]
@renyunkang I tried to follow the code, maybe we just need the `update` verb insted of `patch` verb?

https://github.com/openelb/openelb/blob/e3102a172621993ebf5c224983a98c2d92b7631a/pkg/controllers/lb/nodeproxy.go#L253-L255

https://github.com/openelb/openelb/blob/e3102a172621993ebf5c224983a98c2d92b7631a/pkg/controllers/lb/nodeproxy.go#L419-L422

**Comment by renyunkang** [comment]
> @renyunkang I tried to follow the code, maybe we just need the `update` verb insted of `patch` verb?
> 
> https://github.com/openelb/openelb/blob/e3102a172621993ebf5c224983a98c2d92b7631a/pkg/controllers/lb/nodeproxy.go#L253-L255
> 
> https://github.com/openelb/openelb/blob/e3102a172621993ebf5c224983a98c2d92b7631a/pkg/controllers/lb/nodeproxy.go#L419-L422

You are right, it's possible to remove `patch` permissions, but `patch` has a smaller scope of operation and fewer security risks compared to `update`. An `update` operation can lead to the entire object being replaced, whereas a `patch` operation only modifies specified fields, which presents relatively lower risk.
